types/next/test/next-dynamic-tests.tsx

```diff
@@ -17,26 +17,42 @@ const LoadingComponent: React.StatelessComponent<LoadingComponentProps> = ({
 }) => <p>loading...</p>;
 
 // 1. Basic Usage (Also does SSR)
-const DynamicComponent = dynamic(asyncComponent);
+const DynamicComponent = dynamic(Promise.resolve(MyComponent));
 const dynamicComponentJSX = <DynamicComponent foo="bar" />;
 
+// 1.1 Basic Usage (Loader function, module shape with 'export = Component' / 'module.exports = Component')
+const DynamicComponent2 = dynamic(() => Promise.resolve(MyComponent));
+const dynamicComponent2JSX = <DynamicComponent2 foo="bar" />;
+
+// 1.2 Basic Usage (Loader function, module shape with 'export default Component')
+const DynamicComponent3 = dynamic(() => Promise.resolve({ default: MyComponent }));
+const dynamicComponent3JSX = <DynamicComponent3 foo="bar" />;
+
+// TODO: Work with module shape 'export { Component }'
+
 // 2. With Custom Loading Component
-const DynamicComponentWithCustomLoading = dynamic(asyncComponent, {
+const DynamicComponentWithCustomLoading = dynamic(import('./imports/with-default'), {
+    loading: LoadingComponent
+});
+const dynamicComponentWithCustomLoadingJSX = <DynamicComponentWithCustomLoading foo />;
+
+// 2.1. With Custom Loading Component (() => import('') syntax)
+const DynamicComponentWithCustomLoading2 = dynamic(() => import('./imports/with-default'), {
     loading: LoadingComponent
 });
-const dynamicComponentWithCustomLoadingJSX = <DynamicComponentWithCustomLoading foo="bar" />;
+const dynamicComponentWithCustomLoading2JSX = <DynamicComponentWithCustomLoading2 foo />;
 
 // 3. With No SSR
-const DynamicComponentWithNoSSR = dynamic(asyncComponent, {
+const DynamicComponentWithNoSSR = dynamic(() => import('./imports/with-default'), {
     ssr: false
 });
 
 // 4. With Multiple Modules At Once
 const HelloBundle = dynamic<MyComponentProps>({
     modules: () => {
         const components = {
-            Hello1: asyncComponent,
-            Hello2: asyncComponent
+            Hello1: () => import('./imports/with-default'),
+            Hello2: () => import('./imports/with-default')
```

Comment:
isn't this a different test? and isn't it already covered by `1.x` scenarios?

Reply:
I was thinking of deleting all the tests that were migrated to `next-server` but thought against it. Maybe I should just delete it after all since it's probably useless to duplicate these tests in the `next` package as well.

Reply:
Yes, I already got confused 😬